tests/test_core/test_annotationdb.py

```diff
@@ -0,0 +1,633 @@
+    )
+    got = db._make_sql_query(
+        seq_name=None,
+        bio_type="CDS",
+        identifier="RandomIdentifier",
+        start=None,
+        end=None,
+    )
+    assert got == expected
+
+    """start provided along with identifier and bio_type"""
+    expected = (
+        "SELECT * FROM GENBANK WHERE Type == ? AND Locus_Tag like ? AND Start >= ?",
+        ["CDS", "%RandomIdentifier%", 0],
+    )
+    got = db._make_sql_query(
+        seq_name=None, bio_type="CDS", identifier="RandomIdentifier", start=0, end=None
+    )
+    assert got == expected
+
+    """end provided along with identifier and bio_type"""
+    expected = (
+        "SELECT * FROM GENBANK WHERE Type == ? AND Locus_Tag like ? AND End < ?",
+        ["CDS", "%RandomIdentifier%", 5000],
+    )
+    got = db._make_sql_query(
+        seq_name=None,
+        bio_type="CDS",
+        identifier="RandomIdentifier",
+        start=None,
+        end=5000,
+    )
+    assert got == expected
+
+    """start and end provided along with identifier and bio_type"""
+    expected = (
+        "SELECT * FROM GENBANK WHERE Type == ? AND Locus_Tag like ? AND Start >= ? AND End < ?",
+        ["CDS", "%RandomIdentifier%", 0, 5000],
+    )
+    got = db._make_sql_query(
+        seq_name=None, bio_type="CDS", identifier="RandomIdentifier", start=0, end=5000
+    )
+    assert got == expected
+
+    """all five attributes provided"""
+    expected = (
+        "SELECT * FROM GENBANK WHERE LocusID == ? AND Type == ? AND Locus_Tag like ? AND Start >= ? AND End < ?",
+        ["1", "CDS", "%RandomIdentifier%", 0, 5000],
+    )
+    got = db._make_sql_query(
+        seq_name="1", bio_type="CDS", identifier="RandomIdentifier", start=0, end=5000
+    )
+    assert got == expected
+
+    """check exception when both bio_type and identifier are missing"""
+    import pytest
+
+    with pytest.raises(ValueError):
+        db._make_sql_query(
+            seq_name=None, bio_type=None, identifier=None, start=None, end=None
+        )
+
+    """check exception when both bio_type and identifier are missing, even if other attributes"""
+
+    with pytest.raises(ValueError):
+        db._make_sql_query(
+            seq_name="1", bio_type=None, identifier=None, start=0, end=1000
+        )
+
+    """check exception when only seq_name is provided"""
+
+    with pytest.raises(ValueError):
+        db._make_sql_query(
+            seq_name="1", bio_type=None, identifier=None, start=None, end=None
+        )
+
+    """check exception when only start is provided"""
+
+    with pytest.raises(ValueError):
+        db._make_sql_query(
+            seq_name=None, bio_type=None, identifier=None, start=0, end=None
+        )
+
+    """check exception when only end is provided"""
+
+    with pytest.raises(ValueError):
+        db._make_sql_query(
+            seq_name=None, bio_type=None, identifier=None, start=None, end=1000
+        )
+
+
+if __name__ == "__main__":
```

Comment:
remove the `if __name__` block -- not required. `pytest` will find these tests without.